api/client/service/update.go

```diff
@@ -357,7 +357,12 @@ func removeItems(
 func updateMounts(flags *pflag.FlagSet, mounts *[]mounttypes.Mount) {
 	if flags.Changed(flagMountAdd) {
 		values := flags.Lookup(flagMountAdd).Value.(*MountOpt).Value()
-		*mounts = append(*mounts, values...)
+
+		exist := findAndReplaceMountData(mounts, values)
```

Comment:
```
var mountsByTarget map[string]Mount
for _, mount := range mounts {
  if _, ok := mountsByTarget[mount.Target]; ok {
    return fmt.Errorf("duplicate mount target")
  }

  mountsByTarget[mount] = mount
}

var mounts []Mount

for _, mount := range mountsByTarget {
  mounts = append(mounts, mount)
}

sortMountsBySourceThenTarget(mounts)
```

The find and replace isn't really necessary. Just rebuild the dataset correctly.